src/arrays.js

```diff
@@ -8,33 +8,74 @@ const each = (elements, cb) => {
   // Iterates over a list of elements, yielding each in turn to the `cb` function.
   // This only needs to work with arrays.
   // based off http://underscorejs.org/#each
+  for (let i = 0; i < elements.length; i++) {
+    cb(elements[i], i);
+  }
 };
 
 const map = (elements, cb) => {
   // Produces a new array of values by mapping each value in list through a transformation function (iteratee).
   // Return the new array.
+  const results = [];
+  each(elements, (item) => {
+    results.push(cb(item));
+  });
+  return results;
 };
 
 const reduce = (elements, cb, memo = elements.shift()) => {
   // Combine all elements into a single value going from left to right.
   // Elements will be passed one by one into `cb`.
   // `memo` is the starting value.  If `memo` is undefined then make `elements[0]` the initial value.
+  let i = 0;
+  if (memo === undefined) {
```

Comment:
No need to check if `memo` is undefined in this case, since you're already doing `memo = elements.shift()` as the default parameter to this function. That default assignment only happens when no value for `memo` is passed to this function.